internal/librarian/generate.go

```diff
@@ -86,30 +94,25 @@ func init() {
 	})
 }
 
-func runGenerate(ctx context.Context) error {
-	if err := validateRequiredFlag("api-path", flagAPIPath); err != nil {
-		return err
-	}
-	if err := validateRequiredFlag("api-root", flagAPIRoot); err != nil {
+func runGenerate(ctx context.Context, apiPath, apiRoot, repoRoot, repoURL, secretsProject string, build bool) error {
```

Comment:
This signature is error-prone because there are so many string args in a row. Consider taking a struct.

Reply:
Actually I figured out a way. Did this on a follow up PR https://github.com/googleapis/librarian/pull/535. Let me know what you think.

Reply:
LGTM.